File: Sort_Search/restaurant-cust.py
n = int(input())
a = []

for i in range(n):
    x, y = map(int, input().split())
    a.extend([(x, 1), (y, -1)])

a.sort()

m = 0
cur = 0
i = 0

for _, val in a:
    cur += val
    m = max(cur, m)

print(m)

File: Sort_Search/restaurant-cust.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
    int n;
    std::cin >> n;

    std::vector<std::pair<int, int>> a;

    for (int i = 0; i < n; ++i) {
        int x, y;
        std::cin >> x >> y;
        a.push_back(std::make_pair(x, 1));
        a.push_back(std::make_pair(y, -1));
    }

    std::sort(a.begin(), a.end());

    int m = 0;
    int cur = 0;

    for (const auto& p : a) {
        cur += p.second;
        m = std::max(cur, m);
    }

    std::cout << m << std::endl;

    return 0;
}